lib/facility.rb

```diff
@@ -1,14 +1,65 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services,
+              :registered_vehicles,
+              :collected_fees,
+              :facility_created
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_info)
+    @name = facility_info[:name]
+    @address = facility_info[:address]
+    @phone = facility_info[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
+    @facility_created = []
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    if @services.include?('Vehicle Registration') == true
+      @registered_vehicles << vehicle
+      vehicle.set_registration_date
+      vehicle.give_plate
+        if vehicle.electric_vehicle? == true
+          @collected_fees += 200
+        elsif vehicle.antique? == true
+          @collected_fees += 25
+        else
+          @collected_fees += 100
+        end
+        return [vehicle] 
+    else 
+      return nil
+    end
```

Comment:
Overall, great work with all this logic and including all the side effects we wanted register_vehicle to display. Moving forward, I want us to start thinking about how we can break this logic up into more helper methods, that way each method can adhere to SRP (Single Responsibility Principle).